front/components/assistant/conversation/AssistantQuickEditionMenu.tsx

```diff
@@ -0,0 +1,134 @@
+import {
+  ClipboardIcon,
+  DropdownMenu,
+  Icon,
+  ListAddIcon,
+  ListRemoveIcon,
+  MoreIcon,
+  PencilSquareIcon,
+} from "@dust-tt/sparkle";
+import type { AgentUserListStatus, WorkspaceType } from "@dust-tt/types";
+import {
+  isAgentConfigurationInWorkspace,
+  isAgentConfigurationPublished,
+  isBuilder,
+  isGlobalAgentConfiguration,
+} from "@dust-tt/types";
+import { useContext, useState } from "react";
+
+import { SendNotificationsContext } from "@app/components/sparkle/Notification";
+import { performAgentUserListStatusUpdate } from "@app/lib/client/dust_api";
+import { useAgentConfiguration } from "@app/lib/swr";
+
+interface AssistantQuickEditionMenuProps {
+  agentConfigurationSId: string;
+  owner: WorkspaceType;
+}
+
+export function AssistantQuickEditionMenu({
+  // The `agentConfiguration` cannot be used directly as it isn't dynamically
+  // updated upon user list mutations. This limitation stems from its
+  // propagation method from <ConversationMessage>.
+  agentConfigurationSId,
+  owner,
+}: AssistantQuickEditionMenuProps) {
+  const [isUpdatingList, setIsUpdatingList] = useState(false);
+  const sendNotification = useContext(SendNotificationsContext);
+
+  const { agentConfiguration, mutateAgentConfiguration } =
+    useAgentConfiguration({
+      workspaceId: owner.sId,
+      agentConfigurationId: agentConfigurationSId,
+    });
+
+  if (!agentConfiguration) {
+    return <></>;
+  }
+
+  if (isGlobalAgentConfiguration(agentConfiguration)) {
+    return <></>;
+  }
+
+  const isAgentInWorkspace =
+    isAgentConfigurationInWorkspace(agentConfiguration);
+  const isAgentPublished = isAgentConfigurationPublished(agentConfiguration);
+
+  const isInList = agentConfiguration.userListStatus === "in-list";
```

Comment:
nit: cf below, I'm usually in favor of doing that, but here it's adding code for something that's already clear enough

very nit though :)